tests/run_test.sh

```diff
@@ -86,11 +86,17 @@ for test in "$@"; do
     echo -n "test ${white}$test${off}.in: "
     result=$("$cmd" < "$test.in")
   fi
-  # exit code stored in $?
+  exit_code=$?
 
-  if [ $? -ne $expect ]; then
+  if [ $exit_code -ne $expect ]; then
     # If the exit code is wrong, the test is a failure and --bless won't help
-    echo "${red}failed${off}"; exit_code=1; continue
+    echo "${red}failed${off} (exit code = $exit_code)"; exit_code=1
```

Comment:
I don't know what the exit_code is set for.  The continue forces the next loop, or exits it, if it was the last test. In the first case it is overwritten in line 89.  In the second case it forces the result of this script to 1, indicating a failure.

A failing test in the middle has no impact on the result of the script, only if it is last??

Reply:
oh oops, there is a name clash there. The `exit_code=1` is for the exit_code of the script itself, while the `$exit_code` reference is the exit code of the metamath invocation of the current test. Will fix.